tests/integration/slashing.go

```diff
@@ -665,3 +665,41 @@ func (suite *CCVTestSuite) TestQueueAndSendSlashPacket() {
 	suite.Require().Empty(dataPackets)
 	suite.Require().Len(dataPackets.GetList(), 0)
 }
+
+// TestCISBeforeCCVEstablished tests that the consumer chain doesn't panic or
+// have any undesired behavior when a slash packet is queued before the CCV channel is established.
+// Then once the CCV channel is established, the slash packet should be sent soon after.
+func (suite *CCVTestSuite) TestCISBeforeCCVEstablished() {
+	consumerKeeper := suite.consumerApp.GetConsumerKeeper()
+
+	// Check pending packets is empty
+	pendingPackets := consumerKeeper.GetPendingPackets(suite.consumerCtx())
+	suite.Require().Len(pendingPackets.List, 0)
+
+	consumerKeeper.Slash(suite.consumerCtx(), []byte{0x01, 0x02, 0x3},
+		66, 4324, sdk.MustNewDecFromStr("0.05"), stakingtypes.Downtime)
+
+	// Check slash packet was queued
+	pendingPackets = consumerKeeper.GetPendingPackets(suite.consumerCtx())
+	suite.Require().Len(pendingPackets.List, 1)
+
+	// Pass 5 blocks, confirming the consumer doesn't panic
+	suite.consumerChain.NextBlock()
+	suite.consumerChain.NextBlock()
+	suite.consumerChain.NextBlock()
+	suite.consumerChain.NextBlock()
+	suite.consumerChain.NextBlock()
```

Comment:
Why not just iterate here or write a function to pass time?

```go
func produceEmptyBlocks(suite *CCVTestSuite, blocks int){
  for blocks > 0 {
    suite.consumerChain.NextBlock()
    blocks--
  }
}
```

Reply:
"empty blocks" is kind of true since no txs are submitted. But when we do something like this it's usually a sanity check that beginblock/endblock do or do not trigger some code. Reasoning is usually a case by case basis

For loop makes sense tho https://github.com/cosmos/interchain-security/commit/cb9a1ec93fdbed4c8f09cd4152deac687e46dd45